### ISSUE TITLE
HTTP Rest API Auth failure

### ISSUE THREAD

**Issue by drejkus** [emqtt/emqttd#1274]
#### Environment

- OS: Ubuntu 16.04
- Erlang/OTP: R20/9.0
- EMQ: 2.3-beta4

#### Description

HTTP Rest API will not authenticate any users. I have only MySQL Auth enabled.
test command:
`curl -v --basic -u myuser:mypass -d "qos=0&retain=0&topic=/a/b/c&message=hello from http..." -k http://localhost:8080/api/v2/mqtt/publish`

in error.log:

> 2017-09-26 14:17:55.913 [error] <0.1612.0>@emqttd_http:authorized:139 HTTP Auth failure: username=myuser, reason="User not found"

2.3beta3 works OK

**Comment by turtleDeng** [comment]
@drejkus 
The url of this API has been changed to ‘api/v2/mqtt/publish’ in v2.3-beta.2 release. Read the doc in REST API.
doc is http://emqtt.io/docs/v2/rest.html#messages

Authorized dashboard user is admin/public

```
curl -v --basic -u admin:public -H "Content-Type: application/json" -d '{"topic": "test","payload": "hello","qos": 1,"retain": false,"client_id": "C_1492145414740"}' -k http://localhost:8080/api/v2/mqtt/publish
```

**Comment by drejkus** [comment]
But it works form me in 2.3-beta.3 so it's not this problem, I see it more as a regression bug.  

You also use dashboard user, so was the behavior changed to use dashoard users now? 
If it was I think it'll be good idea to add normal user authentication back (and maybe a way to signal which users to use). At least for me, that is what I was using this for - to create API for normal users from outside MQTT.

**Comment by sincsc** [comment]
where can I config the rest api http basic auth username and password? thank you

**Comment by HJianBo** [comment]
Hi, @drejkus sorry too late reply.

The answer is Yes!  The REST API must be authenticated by Dashboard Users

**Comment by HJianBo** [comment]
Hi, @drejkus sorry too late reply.

The answer is Yes! **The REST API must be authenticated by Dashboard Users**.

And, You can add an admin user (Dashboard User) with the following CLIs:
```
./bin/emqttd_ctl admins add username1 password1
```

In addition, at the next version (v2.3.3 coming soon), you can add some default admin users at the `./etc/plugins/emq_dashboard.conf` configuration file.

**Comment by HJianBo** [comment]
Hi, @drejkus @sincsc The feature had been implemented at v2.3.3 and published on 11 Jan.

Close this issue now. if has any problems reopen it!
